File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int, int> mpp; // Use unordered_map for O(N) speed
        for (int i = 0; i < nums.size(); i++) {
            mpp[nums[i]]++;
        }

        int maxFreq = 0;
        int ans = 0;

        // Correct way to iterate through a map:
        for (auto const& [element, count] : mpp) {
            if (count > maxFreq) {
                maxFreq = count;
                ans = element;
            }
        }
        
        return ans;
    }
};

File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer, Integer> mpp = new HashMap<>();

        for(int i = 0; i < nums.length; i++) {
            if (mpp.containsKey(nums[i])) {
                mpp.put(nums[i], mpp.get(nums[i]) + 1);
            } else {
                mpp.put(nums[i], 1);
            }
        }

        int maxValue = 0;
        int ans = 0;
        for(int key: mpp.keySet()){
            if(mpp.get(key)>maxValue){
                maxValue = mpp.get(key);
                ans = key;
            }
        }

        return ans;
    }
}